board/orders/diplomacy.php

```diff
@@ -155,6 +155,11 @@ protected function viaConvoyCheck()
 	protected function checkConvoyPath($startCoastTerrID, $endCoastTerrID, $mustContainTerrID=false, $mustNotContainTerrID=false) {
 		global $DB;
 
+        if ($this->convoyPath == [-1]) { // API - We are offloading computing the convoy path to the server

```

Comment:
Mmmm... this isn’t the code that computes the path this is the code that checks it. This would let anyone submit a convoy path of -1 and their submitted convoy wouldn’t work. This looks like a security issue unless I’m mistaken; the system needs to check this please.

Reply:
*Their submitted convoy /would/ work, even if it was convoying from Clyde to Turkey with nothing in between, I believe the adjudicator assumes convoy paths are valid.